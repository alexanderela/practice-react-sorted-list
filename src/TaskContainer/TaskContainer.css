.TaskContainer {
	height: 30rem;
	width: 30rem;
  display: flex;
  align-items: center;
  flex-direction: column;
}

li {
	width: 5rem;
}