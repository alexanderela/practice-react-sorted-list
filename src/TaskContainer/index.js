import React from 'react';
import './TaskContainer.css';

const TaskContainer = ({ tasks }) => {
	const taskCards = tasks.map(task => {
		return <li key={task.id}>{task.task}</li>
	})

	return(
		<ol className='TaskContainer'>
			{taskCards}
		</ol>
	)
}

export default TaskContainer;