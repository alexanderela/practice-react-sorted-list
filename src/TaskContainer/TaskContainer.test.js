import React from 'react';
import ReactDOM from 'react-dom';
import TaskContainer from './';
import { shallow } from 'enzyme';

describe('TaskContainer', () => {
	let wrapper;
	let mockTasks;

	beforeEach(() => {
		mockTasks = [
			{
				task: 'Feed cat',
				error: '',
				id: 123
			},
			{
				task: 'Walk dog',
				error: '',
				id: 456
			},
			{
				task: 'Make weekend plans',
				error: '',
				id: 789
			},
		]
		wrapper = shallow(<TaskContainer tasks={mockTasks}/>)
	})

	it('should render like snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('', () => {})
})