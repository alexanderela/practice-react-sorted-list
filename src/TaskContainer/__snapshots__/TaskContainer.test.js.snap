// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskContainer should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TaskContainer
    tasks={
      Array [
        Object {
          "error": "",
          "id": 123,
          "task": "Feed cat",
        },
        Object {
          "error": "",
          "id": 456,
          "task": "Walk dog",
        },
        Object {
          "error": "",
          "id": 789,
          "task": "Make weekend plans",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          Feed cat
        </li>,
        <li>
          Walk dog
        </li>,
        <li>
          Make weekend plans
        </li>,
      ],
      "className": "TaskContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "123",
        "nodeType": "host",
        "props": Object {
          "children": "Feed cat",
        },
        "ref": null,
        "rendered": "Feed cat",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "456",
        "nodeType": "host",
        "props": Object {
          "children": "Walk dog",
        },
        "ref": null,
        "rendered": "Walk dog",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "789",
        "nodeType": "host",
        "props": Object {
          "children": "Make weekend plans",
        },
        "ref": null,
        "rendered": "Make weekend plans",
        "type": "li",
      },
    ],
    "type": "ol",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            Feed cat
          </li>,
          <li>
            Walk dog
          </li>,
          <li>
            Make weekend plans
          </li>,
        ],
        "className": "TaskContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "123",
          "nodeType": "host",
          "props": Object {
            "children": "Feed cat",
          },
          "ref": null,
          "rendered": "Feed cat",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "456",
          "nodeType": "host",
          "props": Object {
            "children": "Walk dog",
          },
          "ref": null,
          "rendered": "Walk dog",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "789",
          "nodeType": "host",
          "props": Object {
            "children": "Make weekend plans",
          },
          "ref": null,
          "rendered": "Make weekend plans",
          "type": "li",
        },
      ],
      "type": "ol",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
